newIDE/app/src/Utils/i18n/language.js

```diff
@@ -0,0 +1,49 @@
+const rtlLanguages = new Set([
+  /******* Hebrew scripts *******/
+  'he'/* Hebrew */,
+  'ji'/* Yiddish */,
+  'yi'/* Yiddish */,
+  
+  /******* Arabic scripts *******/
+  'ar'/* Arabic */,
+  'az'/* Azerbaijani */,
+  'jv'/* Javanese */,
+  'ks'/* Kashmiri */,
+  'kk'/* Kazakh */,
+  'ku'/* Kurdish */,
+  'ms'/* Malay */,
+  'ml'/* Malayalam */,
+  'ps'/* Pashto */,
+  'fa'/* Persian */,
+  'pa'/* Punjabi */,
+  'sd'/* Sindhi */,
+  'so'/* Somali */,
+  'tk'/* Turkmen */,
+  'ug'/* Uighur */,
+  'ur'/* Urdu */,
+]);
+
+export default class Language {
+  constructor(name) {
+    this._name = '--';
+    if (typeof name === 'string') {
+      this._name = /^(\w{2})?/.exec(name)[1] ?? this._name;
+    }
+  }
+  
+  get is_ltr() {
+    return !rtlLanguages.has(this._name);
+  }
+  
+  get name() {
+    return this._name;
+  }
+  
+  static from(lang) {
+    return typeof lang === 'object' ? lang : new Language(lang);
+  }
+  
+  static is_ltr(lang) {
+    return Language.from(lang)?.is_ltr;
+  }
```

Comment:
I find this approach overly complex: there is a constructor, a static constructor (from), 2 `typeof` (that is better to be avoided in code to favor static typing), a static method.

It seems that the class could be replaced by a call to `Set.has`:

```suggestion
export const isLtr = (lang: string) => !rtlLanguages.has(lang);
```